### PR TITLE
Added documentation to portal routes & fixed small issues

### PR DESCRIPTION
**Pull request by matthewrh**
Changes:
- Added documentation to routes
- Fixed issue with portal login
- Fixed a couple of typos
- Improved consistency

Notes:
- Password reset tokens do not become stale after use. This should probably be looked at.

### CODE REVIEW

File: server/portal/creation.py [hunk 1]
```diff
@@ -190,21 +232,28 @@ def approve_post():
     if account.email != 'pennappslabs@gmail.com':
         return jsonify({'error': 'This account does not have permission to issue post approval decisions.'}), 400
 
-    approved = bool(account.form.get('approved'))
+    approved = bool(request.form.get('approved'))
```
Comment (joshdoman) [inline server/portal/creation.py]: Good catch.

File: server/portal/account.py [hunk 2]
```diff
@@ -68,7 +94,15 @@ def get_account_info():
...
 
-# Request password reset
+"""
+Endpoint: /portal/account/reset/request
+HTTP Methods: POST
+Response Formats: JSON
+Content-Type: application/x-www-form-urlencoded 
+Parameters: email
+
+Request password reset token
```
Comment (joshdoman) [inline server/portal/account.py]: Add "Link with reset token is sent to the account's email"

File: server/portal/account.py [hunk 3]
```diff
@@ -37,10 +56,10 @@ def login():
     if any(x is None for x in [email, password]):
         return jsonify({'error': 'Parameter is missing'}), 400
 
-    pw_hash = bcrypt.generate_password_hash(password)
-    account = PostAccount.query.filter(PostAccount.email == email and bcrypt.check_password_hash(pw_hash,
-                                                                                                 password)).first()
-    if account:
+    account = PostAccount.query.filter(PostAccount.email == email).first()
+    pw_hash = bcrypt.check_password_hash(account.encrypted_password, password)
```
Comment (joshdoman) [inline server/portal/account.py]: Check ```if account:```. We can't assume the user passes a valid email. Also, change ```pw_hash``` to a more descriptive variable name. Maybe ```is_correct_password```?

File: server/portal/creation.py [hunk 4]
```diff
@@ -176,6 +208,16 @@ def save_image():
     return jsonify({'image_url': aws_url})
 
 
+"""
+Endpoint: /portal/post/approve
+HTTP Methods: POST
+Response Formats: JSON
+Content-Type: application/x-www-form-urlencoded
```
Comment (joshdoman) [inline server/portal/creation.py]: Remove trailing whitespace and fix all other Flake8 errors. Can be found here: https://circleci.com/gh/pennlabs/labs-api-server/37?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: server/portal/posts.py [hunk 5]
```diff
@@ -72,9 +80,17 @@ def get_posts():
...
+HTTP Methods: GET
+Response Formats: JSON
+Parameters: account_id
+
+Returns post information by post ID
+"""
 @app.route('/portal/post/<int:post_id>', methods=['GET'])
 def get_post(post_id):
-    account_id = request.args.get('account')
+    account_id = request.args.get('account_id')
```
Comment (joshdoman) [inline server/portal/posts.py]: Will need to update penn-mobile-portal after making this change.

File: server/portal/account.py (hunk 3 above)
Comment (matthewrh) [reply server/portal/account.py]: Was already checking for valid email. Renamed password check.

File: server/portal/posts.py (hunk 5 above)
Comment (matthewrh) [reply server/portal/posts.py]: Reversed these changes to their initial state.

**Pull request merged**